### PR TITLE
Boost QLV

### PR DESCRIPTION
**Pull request by laerreal**
This patch series increases Qemu Log Viewer operation performance.
For 100MiB - 1GiB logs mainly.

### v2
- fix typos

### CODE REVIEW

File: qlv.py [hunk 1]
```diff
@@ -45,14 +45,28 @@
...
 
 # Instructions Treeview styles
 STYLE_DEFAULT = tuple()
 STYLE_DIFFERENCE = ("difference",)
 STYLE_FIRST = ("first",)
 
-class InstructionsTreeview(VarTreeview):
+# Instructions tree view shows only few instructions in backend Tk Treeview.
+# Showed instructions interval is called "window".
+# The window is automaticlay shifted when necessery by adding/removing rows
```
Comment (BDanAnd) [inline qlv.py]: automatically, necessary

File: qlv.py [hunk 2]
```diff
@@ -45,14 +45,28 @@
...
+# beyond the window.
+# Curretly 1024 instructions are showed to user.
+TV_WINDOW_SIZE = 1 << 10
+
+# TODO: If user see all showed instructions, scrolling does not works.
+# For bigger dispalays we shoud either catch mouse whell, arrow buttons and
```
Comment (BDanAnd) [inline qlv.py]: displays, should, wheel

File: qlv.py [hunk 3]
```diff
@@ -45,14 +45,28 @@
...
 STYLE_DEFAULT = tuple()
 STYLE_DIFFERENCE = ("difference",)
 STYLE_FIRST = ("first",)
 
-class InstructionsTreeview(VarTreeview):
+# Instructions tree view shows only few instructions in backend Tk Treeview.
+# Showed instructions interval is called "window".
+# The window is automaticlay shifted when necessery by adding/removing rows
+# beyond the window.
+# Curretly 1024 instructions are showed to user.
```
Comment (BDanAnd) [inline qlv.py]: Currently

File: qlv.py [hunk 4]
```diff
@@ -74,6 +95,241 @@ def __init__(self, master, **kw):
...
+
+    def see_instruction(self, idx):
+        # We trying to keep user view in the middle of the window
+        shift = int(idx - TV_WINDOW_HALF - self._window_start)
+
+        self._shift_window(shift)
+
+        VarTreeview.yview(self, idx - self._window_start)
+        # Previous command not always results in _yscrollcommand feedback.
+        # But the feedback mast always be passed to an outer observer.
```
Comment (BDanAnd) [inline qlv.py]: must

File: qemu/qlog.py [hunk 5]
```diff
@@ -90,6 +90,8 @@ def __str__(self):
         return str(self.in_instr)
 
 
+class undefined: pass
+
```
Comment (BDanAnd) [inline qemu/qlog.py:94]: maybe 2 new lines?

File: qlv.py [hunk 6]
```diff
@@ -74,6 +95,241 @@ def __init__(self, master, **kw):
...
+        new_start = max(0,
+            min(len(all_insts) - TV_WINDOW_SIZE, cur_start + shift)
+        )
+        if new_start == cur_start:
+            return
+
+        actual_shift = new_start - cur_start
+
+        if DEBUG_INST_TV:
+            print("Shifthing window to %d (%d)" % (new_start, actual_shift))
```
Comment (BDanAnd) [inline qlv.py]: Shifting

File: qlv.py [hunk 7]
```diff
@@ -74,6 +95,241 @@ def __init__(self, master, **kw):
...
+
+        current_items = self.get_children()
+
+        _insert = self._insert_instruction_row
+
+        if actual_shift > 0:
+            self.delete(*current_items[:actual_shift])
+
+            new_inst_limit = new_start + TV_WINDOW_SIZE
+            # actual_shift can be bigger than windows size (absolute value)
```
Comment (BDanAnd) [inline qlv.py]: window

File: qemu/qlog.py (hunk 5 above)
Comment (laerreal) [reply qemu/qlog.py:94]: No, this class is only used in `TBCache`. It's actually a constant.

**Pull request merged**